pkgs/development/compilers/mozart/patch-limits.diff

```diff
@@ -0,0 +1,10 @@
+--- mozart2-2.0.1.orig/vm/vm/main/core-forward-decl.hh	2018-09-05 03:16:51.000000000 +0200
++++ mozart2-2.0.1/vm/vm/main/core-forward-decl.hh	2021-08-03 10:31:50.281476430 +0200
+@@ -36,6 +36,7 @@
+ #include <cstdint>
+ #include <ostream>
+ #include <functional>
++#include <limits>
+ #include <memory>
+ #include <cassert>
+ 
```

Comment:
I think such a fix deserves an upstream PR rather than (or together with) a nixpkgs fix. Unless the issue is specific to the nixpkgs layout ?

Reply:
I really don't known who the fault is... I tried various approaches but none worked, only this one.

Without the patch, we get many errors like this one

```
/build/mozart2-2.0.1-Source/vm/vm/main/smallint-decl.hh:42:17: error: no member named 'numeric_limits' in namespace 'std'
    return std::numeric_limits<nativeint>::min();
           ~~~~~^
```

Reply:
Could it be https://www.gnu.org/software/gcc/gcc-11/porting_to.html#header-dep-changes ? That would have been introduced by gcc 11. Maybe the proper fix is to keep building mazart2 with gcc10 (or gcc10Compat ?) while this is not fixed upstream. And upstream gets very few updates ;-).

Reply:
(from discussing with @layus) It's not directly related to gcc headers, because clang is used to build this package, but actually llvm reuses the headers from gcc, so it is related indirectly after all.